src/audio/code_8019AF00.c

```diff
@@ -5089,10 +5445,22 @@ void Audio_ResetData(void);
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A46F8.s")
 
-f32 sSfxOriginalPos[] = { 0.0f, 0.0f, 0.0f };
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A4748.s")
+// used for z_obj_sound and z_en_gk
+void Audio_PlaySfx_AtFixedPos(Vec3f* pos, u16 sfxId) {
+    static f32 sSfxOriginalPos[] = { 0.0f, 0.0f, 0.0f };
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A479C.s")
+    for (i = 0; i < ARRAY_COUNT(sSfxOriginalPos); i++) {
+        sSfxOriginalPos[i] = ((f32*)pos)[i];
+    }
+
+    Audio_PlaySfx_AtPos((Vec3f*)sSfxOriginalPos, sfxId);
+}
```

Comment:
Since it still matches, tbh I'd prefer `sSfxOriginalPos` to be a Vec3f, and just cast in the loop (like pos).
```
void Audio_PlaySfx_AtFixedPos(Vec3f* pos, u16 sfxId) {
    static Vec3f sSfxOriginalPos = { 0.0f, 0.0f, 0.0f };
    s32 i;

    for (i = 0; i < 3; i++) {
        ((f32*)&sSfxOriginalPos)[i] = ((f32*)pos)[i];
    }

    Audio_PlaySfx_AtPos(&sSfxOriginalPos, sfxId);
}

```
You could also create f32 pointer temps as well if you want
```
void Audio_PlaySfx_AtFixedPos(Vec3f* pos, u16 sfxId) {
    static Vec3f sSfxOriginalPos = { 0.0f, 0.0f, 0.0f };
    s32 i;
    f32* tmp1 = (f32*)&sSfxOriginalPos;
    f32* tmp2 = (f32*)pos;

    for (i = 0; i < 3; i++) {
        tmp1[i] = tmp2[i];
    }

    Audio_PlaySfx_AtPos(&sSfxOriginalPos, sfxId);
}
```

Reply:
That last one seems best to me, thanks